simd/unit_tests/include/SIMDTesting_Utilities.hpp

```diff
@@ -49,9 +49,9 @@ class kokkos_checker {
     // double and a and b end up being different in long double but have the
     // same value when casted to float or double. (see
     // https://gcc.gnu.org/bugzilla/show_bug.cgi?id=323#c109)
-    T const volatile va = a;
-    T const volatile vb = b;
-    if (va != vb)
+    T const volatile* const volatile pa = &a;
+    T const volatile* const volatile pb = &b;
+    if (*pa != *pb)
```

Comment:
Why did you change this?

Reply:
Yes, in fact, Trévis had already placed in a block of comment describing the reason we placed in this `ifdef` block.
```
    // This is needed to work around a bug where the comparison fails because it
    // is done on the x87 fpu (which is the default for 32 bit gcc) in long
    // double and a and b end up being different in long double but have the
    // same value when casted to float or double. (see
    // https://gcc.gnu.org/bugzilla/show_bug.cgi?id=323#c109)
 ```